src/client.rs

```diff
@@ -49,6 +49,28 @@ macro_rules! impl_batch_call {
     }};
 }
 
+/// A trait for [`ToSocketAddrs`](https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html) that
+/// can also be turned into a domain. Used when an SSL client needs to validate the server's
+/// certificate.
+pub trait ToSocketAddrsDomain: ToSocketAddrs {
+    /// Returns the domain, if present
+    fn domain(&self) -> Option<&str> {
+        None
+    }
+}
+
+impl ToSocketAddrsDomain for &str {
+    fn domain(&self) -> Option<&str> {
+        self.splitn(2, ':').into_iter().next()
```

Comment:
nit: clippy says you can remove into_iter()

Reply:
fixed in 78a8a7c